### PR TITLE
[L1T] phase-2, update L1 emulation objects

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

Porting leftovers of https://github.com/cms-l1t-offline/cmssw/pull/996 in cms-l1t-offline. Mostly changes to SimL1Emulator

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

File: L1Trigger/Configuration/python/SimL1Emulator_cff.py [hunk 1]
```diff
@@ -90,17 +94,54 @@
...
 _phase2_siml1emulator.add(l1EGammaEEProducer)
 
+# Barrel and EndCap CaloJet/HT
+# ########################################################################
+# ----    Produce the calibrated tower collection combining Barrel, HGCal, HF
+from L1Trigger.L1CaloTrigger.L1TowerCalibrationProducer_cfi import *
+L1TowerCalibration = L1TowerCalibrationProducer.clone(
+  L1HgcalTowersInputTag = cms.InputTag("hgcalTowerProducer","HGCalTowerProcessor",""),
+  #l1CaloTowers = cms.InputTag("L1EGammaClusterEmuProducer","","")
+  l1CaloTowers = cms.InputTag("L1EGammaClusterEmuProducer","L1CaloTowerCollection","")
```
Comment (perrotta) [inline L1Trigger/Configuration/python/SimL1Emulator_cff.py:104]:
```suggestion
  L1HgcalTowersInputTag = ("hgcalTowerProducer","HGCalTowerProcessor",""),
  l1CaloTowers = ("L1EGammaClusterEmuProducer","L1CaloTowerCollection","")
```

File: L1Trigger/Configuration/python/SimL1Emulator_cff.py [hunk 2]
```diff
@@ -90,17 +94,54 @@
...
+)
+# ----    Produce the L1CaloJets
+from L1Trigger.L1CaloTrigger.L1CaloJetProducer_cfi import *
+L1CaloJet = L1CaloJetProducer.clone (
+    l1CaloTowers = cms.InputTag("L1TowerCalibration","L1CaloTowerCalibratedCollection",""),
+    L1CrystalClustersInputTag = cms.InputTag("L1EGammaClusterEmuProducer", "","")
```
Comment (perrotta) [inline L1Trigger/Configuration/python/SimL1Emulator_cff.py:110]:
```suggestion
    l1CaloTowers = ("L1TowerCalibration","L1CaloTowerCalibratedCollection",""),
    L1CrystalClustersInputTag = ("L1EGammaClusterEmuProducer", "","")
```

File: L1Trigger/Configuration/python/SimL1Emulator_cff.py [hunk 3]
```diff
@@ -90,17 +94,54 @@
...
+)
+# ----    Produce the CaloJet HTT Sums
+from L1Trigger.L1CaloTrigger.L1CaloJetHTTProducer_cfi import *
+L1CaloJetHTT = L1CaloJetHTTProducer.clone(
+    BXVCaloJetsInputTag = cms.InputTag("L1CaloJet", "CaloJets")
```
Comment (perrotta) [inline L1Trigger/Configuration/python/SimL1Emulator_cff.py:115]:
```suggestion
    BXVCaloJetsInputTag = ("L1CaloJet", "CaloJets") 
```

File: L1Trigger/Configuration/python/SimL1Emulator_cff.py [hunk 4]
```diff
@@ -90,17 +94,54 @@
...
+
 # ########################################################################
 # Phase-2 L1T - TrackTrigger dependent modules
 # ########################################################################
+from L1Trigger.L1TTrackMatch.L1GTTInputProducer_cfi import *
+from L1Trigger.VertexFinder.VertexProducer_cff import *
+L1VertexFinder = VertexProducer.clone()
+L1VertexFinderEmulator = VertexProducer.clone()
+L1VertexFinderEmulator.VertexReconstruction.Algorithm = "fastHistoEmulation"
+L1VertexFinderEmulator.l1TracksInputTag = cms.InputTag("L1GTTInputProducer","Level1TTTracksConverted")
```
Comment (perrotta) [inline L1Trigger/Configuration/python/SimL1Emulator_cff.py:131]:
```suggestion
L1VertexFinderEmulator.l1TracksInputTag = ("L1GTTInputProducer","Level1TTTracksConverted")
```

File: L1Trigger/Configuration/python/SimL1Emulator_cff.py [hunk 5]
```diff
@@ -112,43 +153,102 @@
...
+_phase2_siml1emulator.add( L1TkMuonsTP )
+_phase2_siml1emulator.add( L1TkGlbMuons )
+
+
+# Emulated GMT Muons (Tk + Stub, Tk + MuonTFT, StandaloneMuon)
+# ########################################################################
+from L1Trigger.Phase2L1GMT.gmt_cfi  import *
+L1TkStubsGmt = gmtStubs.clone()
+L1TkMuonsGmt = gmtMuons.clone(
+    srcStubs  = cms.InputTag('L1TkStubsGmt')
```
Comment (perrotta) [inline L1Trigger/Configuration/python/SimL1Emulator_cff.py:170]:
```suggestion
    srcStubs  ='L1TkStubsGmt'
```

File: L1Trigger/Configuration/python/SimL1Emulator_cff.py [hunk 6]
```diff
@@ -112,43 +153,102 @@
...
+# ########################################################################
+from L1Trigger.L1TTrackMatch.L1TrackJetProducer_cfi import *
+from L1Trigger.L1TTrackMatch.L1TrackFastJetProducer_cfi import *
+from L1Trigger.L1TTrackMatch.L1TrackerEtMissProducer_cfi import *
+from L1Trigger.L1TTrackMatch.L1TkHTMissProducer_cfi import *
+# make the input tags consistent with the choice L1VertexFinder above
+L1TrackJets.L1PVertexCollection  = cms.InputTag("L1VertexFinder", L1VertexFinder.l1VertexCollectionName.value())
+L1TrackJetsExtended.L1PVertexCollection  = cms.InputTag("L1VertexFinder", L1VertexFinder.l1VertexCollectionName.value())
+L1TrackerEtMiss.L1VertexInputTag = cms.InputTag("L1VertexFinder", L1VertexFinder.l1VertexCollectionName.value())
+L1TrackerEtMissExtended.L1VertexInputTag = cms.InputTag("L1VertexFinder", L1VertexFinder.l1VertexCollectionName.value())
```
Comment (perrotta) [inline L1Trigger/Configuration/python/SimL1Emulator_cff.py:187]:
```suggestion
L1TrackJets.L1PVertexCollection  = ("L1VertexFinder", L1VertexFinder.l1VertexCollectionName.value())
L1TrackJetsExtended.L1PVertexCollection  = ("L1VertexFinder", L1VertexFinder.l1VertexCollectionName.value())
L1TrackerEtMiss.L1VertexInputTag = ("L1VertexFinder", L1VertexFinder.l1VertexCollectionName.value())
L1TrackerEtMissExtended.L1VertexInputTag = ("L1VertexFinder", L1VertexFinder.l1VertexCollectionName.value())
```

**Review by perrotta — COMMENTED** [review]
Besides removing the explicit types for cloned parameters (if they already exist in the original config): was any effect expected on L1T outputs from this PR? None is visible in the DQM plots.
By the way: this is not the fix supposed to be applied for the puppi related distributions which were found wrong in the previous big L1T Phasae2 PR, am I correct?

**Comment by qliphy** [comment]
@cms-sw/l1-l2 Please have a look at above comments

**Pull request closed without merging**